packages/app-mobile/components/useStyles.ts

```diff
@@ -0,0 +1,12 @@
+import { StyleSheet } from 'react-native';
+import { Style } from './global-style';
+import { themeStyle } from './global-style';
+
+const useStyles = (stylingFunction: (theme: Style)=> Style, themeId?: number) => {
+	const theme = themeStyle(themeId);
+	const styles = stylingFunction(theme);
+
+	return StyleSheet.create(styles);
+};
+
+export default useStyles;
```

Comment:
Hmm, I see you made a more generic reusable function to set styles, but I don't think it's necessary. I'm concerned about the indirection and the fact that stylingFunction should be defined using useCallback or else it won't work properly. Just put the useStyle hook directly with the component - it's a bit of boiler plate but it's not too bad and will make it clearer in terms of dependencies.

Reply:
Alright, noted.